api/v1alpha3/awsmachine_webhook.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"errors"
+	"reflect"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+func (r *AWSMachine) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1alpha3-awsmachine,mutating=false,failurePolicy=fail,groups=infrastructure.cluster.x-k8s.io,resources=awsmachines,versions=v1alpha3,name=vawsmachine.kb.io
```

Comment:
It looks like we can override the odd `vawsmachine.kb.io` naming through this annotation...  Could we name it `validating.awsmachine.x-k8s.io` here instead? @ncdc @vincepri thoughts?

Reply:
Probably would be clearer. We'll have to remember to make this change for every webhook.

Reply:
@ncdc we could add a quick script to validate the default naming isn't used for CI. Might be able to even use something as naive as: `if git --no-pager grep kb.io; then echo "kb.io type prefix found"; exit 1; fi`

Reply:
I created a PR for renaming the validation webhook for AWSMachineTemplate here, and we can discuss more on that PR: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1219